bundles/org.dataflowanalysis.analysis/src/org/dataflowanalysis/analysis/core/FlowGraph.java

```diff
@@ -9,17 +9,32 @@
  * implementation of the flow graph
  */
 public abstract class FlowGraph {
-    protected final ResourceProvider resourceProvider;
-    private final List<? extends AbstractPartialFlowGraph> partialFlowGraphs;
+    protected ResourceProvider resourceProvider;
+    private List<? extends AbstractPartialFlowGraph> partialFlowGraphs;
+    
+    
+    /**
+     * Creates a new flow graph. Initialize should be called before the FlowGraph is used.
+     */
+    protected FlowGraph() {
+    }
+    
+    /**
+     * Initializes the used resourceProvider and the partialFlowGraphs by calling {@link #findPartialFlowGraphs()}.
+     * @param resourceProvider the resourceProvider
+     */
+    protected void initialize(ResourceProvider resourceProvider) {
+    	this.resourceProvider = resourceProvider;
+    	this.partialFlowGraphs = this.findPartialFlowGraphs();
+    }
```

Comment:
I don't like that a flow graph may contain null values, but I also understand that this behavior may be required.
Could you quickly outline why this change is required?

Reply:
The FlowGraph calls findPartialFlowGraphs() in the constructor. This mehthod is specified by the subclass (PCMFlowGraph). 
Since I want to allow the use of different factories for the creation of vertices, these factories have to be used in the creation process. findPartialFlowGraphs() calls a PCMPartialFlowGraphFinder (which I have modiefied to allow the creation with specified factories). In the current implementation the PCMPartialFlowGraphFinder is only dependent on the PCMResourceProvider which is initialized in the superclass FlowGraph.

Now the problem arrises when a new dependency is needed / wanted for the creation of a PCMPartialFlowGraphFinder (the factory). Since the super()-constructor has to be called first before initializing variables in the PCMFlowGraph constructor, the factory is not accessable in the findPartialFlowGraphs()-method. Further a PCMPartialFlowGraphFinder which should be used cannot be specified for the same reason.

Therefore, I added the possility to create a FlowGraph without initialization. This allows the subclass to first initialize the needed factories before calling for the initialization in the constructor.